.spin/cmds.py

```diff
@@ -0,0 +1,124 @@
+import hashlib
+import subprocess
+from pathlib import Path
+
+import click
+import spin
+
+
+DEFAULT_HASH_FILE = Path(".lintbin/.lintrunner.sha256")
+
+DEFAULT_FILES_TO_HASH = [
+    "requirements.txt",
+    "pyproject.toml",
+    ".lintrunner.toml",
+]
+
+
+def file_digest(file, algorithm: str):
+    try:
+        return hashlib.file_digest(file, algorithm)
+    except AttributeError:
+        pass  # Fallback to manual implementation below
+    hash = hashlib.new(algorithm)
+    while chunk := file.read(8192):
+        hash.update(chunk)
+    return hash
+
+
+def _hash_file(file):
+    with open(file, "rb") as f:
+        hash = file_digest(f, "sha256")
+    return hash.hexdigest()
+
+
+def _hash_files(files):
+    hashes = {file: _hash_file(file) for file in files}
+    return hashes
+
+
+def _read_hashes(hash_file: Path):
+    if not hash_file.exists():
+        return {}
+    with hash_file.open("r") as f:
+        lines = f.readlines()
+    hashes = {}
+    for line in lines:
+        hash = line[:64]
+        file = line[66:].strip()
+        hashes[file] = hash
+    return hashes
+
+
+def _updated_hashes():
+    new_hashes = _hash_files(DEFAULT_FILES_TO_HASH)
+    old_hashes = _read_hashes(DEFAULT_HASH_FILE)
+    if new_hashes != old_hashes:
+        return new_hashes
+    return None
+
+
+@click.command()
+def setup_lint():
```

Comment:
Part of the setup should be to make sure the type stubs are fresh.
You most likely want these https://github.com/pytorch/pytorch/blob/bfc0ba4af97a1169d6fee5692fae34051b750a12/.github/scripts/lintrunner.sh#L27-L32

Reply:
These sound like also useful in themselves, so how about one or more separate regenerate commands, that also may be called automatically as part of the linting?